pkg/apis/kops/util/labels.go

```diff
@@ -25,9 +25,15 @@ func GetNodeRole(node *v1.Node) string {
 	if _, ok := node.Labels["node-role.kubernetes.io/master"]; ok {
 		return "master"
 	}
+	if _, ok := node.Labels["node-role.kubernetes.io/control-plane"]; ok {
+		return "control-plane"
+	}
 	if _, ok := node.Labels["node-role.kubernetes.io/node"]; ok {
 		return "node"
 	}
+	if _, ok := node.Labels["node-role.kubernetes.io/api-server"]; ok {
```

Comment:
This one is hard to settle. The API Server is part of the control plane, but here we're projecting to only a single role, so even though a node could be labeled with both, we can only take the first one that matches.

Reply:
This is the crux I think.
Should we add this role also to the "main" CP nodes? They would then now have three roles. four if they are spots.

Curious what other k8s implementations do here.

Reply:
I think we should, as this style of annotation deliberately accommodates any number of roles per node. It would be possible to have a machine run, say, the controller manager and scheduler, and _not_ run the API server, so these multiple roles occurring together make sense to clarify what's running there.

Reply:
Well. _This_ particular part of the code doesn't really deal with multiple roles. It is just used by some funny parts of `dns-controller`